lacework/go-sdk
# feat(RAIN-91334): allow Rego queries to be created and retrieved

## Summary

Allow a Rego query to be created from a given yaml file with lacework query create -f MyRego.yaml, and it can be returned via lacework query show command.  The yaml file have three fields:
```
---
queryId: TestRegoQueryViaCli2
queryLanguage: Rego
queryText: |-
  package lacework.policies.lacework_global_134
  import future.keywords
  import data.lacework
  import data.lacework.assessment
  source := lacework.aws.cfg.list("s3", "list-buckets")
  assess := assessment.violation(input, "just because")
```
Now the input and output of query creation need to have a new field `queryLanguage` which is optional.

## How did you test this change?
* Run cli locally in GoLand with program arguments `query create -f rego_query.yaml`,  got response:
```
The query TestRegoQueryViaCli2 was deleted.
```
* Run the same command again with an invalid queryLanguage, got a failure:
```
ERROR unable to create query: 
  [POST] https://dev8.dev8.corp.lacework.net/api/v2/Queries
  [400] Problem parsing JSON, the value for key queryLanguage is of the wrong type
```
* Run cli locally in Goland with program arguments  `query show TestRegoQueryViaCli2 --json`,  got response as expected:
```
{
  "lastUpdateTime": "2023-11-29T06:48:29.000Z",
  "lastUpdateUser": "kun.chen@lacework.net",
  "owner": "kun.chen@lacework.net",
  "queryId": "TestRegoQueryViaCli2",
  "queryLanguage": "Rego",
  "queryText": "package lacework.policies.lacework_global_134\nimport future.keywords\nimport data.lacework\nimport data.lacework.assessment\nsource := lacework.aws.cfg.list(\"s3\", \"list-buckets\")\nassess := assessment.violation(input, \"just because\")",
  "resultSchema": null
}

```

## Issue

https://lacework.atlassian.net/browse/RAIN-91334
